### DISCUSSION TITLE
Not sure where to go from here: "Log in to Grafana and Add Influx as a source"

### DISCUSSION THREAD

**Discussion by bobcov** [jblance/mpp-solar#490] [General]
Hi,
    Do I need to do something explicitly to add influx as a source or was it actually taken care of in earlier steps?
https://github.com/jblance/mpp-solar/blob/master/docs/MQTT_Influx_Grafana.md
I logged in, no errors, influx is listening on the right port, I get this from "show measurements":
cpu
disk
diskio
kernel
mem
processes
swap
system

The instructions seem to end at this stage.  What should happen next with seeing anything in Grafana?

**Comment by jblance** [comment]
you need to check each step / system (post logs / debug if you want more advice as your post doesnt give enough information)
so check the messages are getting to mqtt broker
check the telegraf logs
check influx (as above which shows no data has been loaded from mppsolar)

**Reply by bobcov** [reply]
I've decided to start over because i never had the right version of Python installed and there could be complications due to that error. I don't know, for example, if the unknown connection error had anything to do with that for mqtt, also I had rights issues with mpp-solar getting the serial port unless I opened with sudo and that didn't work when it was running as a service, etc. Will mpp-solar know automatically that python 3.12 is available after I reinstall it? I also have an identical box which I am building from the ground up so that if this one doesn't work I can check to see if the same problem is on the new one.

Following the directives, on the new machine I have a repeat of the permissions issue. 
 mpp-solar -q localhost -c QID -o influx2_mqtt
2024-05-03 11:14:18,014:WARNING:serialio:send_and_receive@33: Serial read error: [Errno 13] could not open port /dev/ttyUSB0: [Errno 13] Permission denied: '/dev/ttyUSB0'

The choices for installation are not explicitly clear to me in their meanings. Of these choices, i used [systemd:
pip install mppsolar (minimal install), or
    pip install mppsolar[api] (install server stuff), or
    pip install mppsolar[ble] (for Bluetooth support aka jkbms), or
    pip install mppsolar[mongo] (for MongoDB output), or
    pip install mppsolar[pgsql] (for PostgreSQL output), or
    pip install mppsolar[push] (for Prometheus PushGateway output), or
    pip install mppsolar[systemd] (for --daemon on Linux), or

But if i selected [systemd], does it mean I cannot have Prometheus or any of the other outputs?
Also unclear why there is this warning, which seems to indicate that choosing [systemd] is not really a method for obtaining that functionality.: WARNING: mppsolar 0.15.62 does not provide the extra 'systemd'

**Reply by bobcov** [reply]
So, not using mpp-solar as a service and making slow progress. I get this connection when I send QPIGS command:
1714998072: New connection from 127.0.0.1 on port 1883.
1714998072: New client connected from 127.0.0.1 as auto-901C77A0-EC45-4574-FA0E-F0016F189E2B (p2, c1, k60).
1714998072: Client auto-901C77A0-EC45-4574-FA0E-F0016F189E2B disconnected.
Telegraf seems reasonably happy with its connection:
2024-05-06T12:35:31Z I! [inputs.mqtt_consumer] Connected [tcp://127.0.0.1:1883]

So, it seems to be talking.  But  the influx "show measurements" command shows only mpp-solar in addition to 
cpu
disk
diskio
kernel
mem
mpp-solar
processes
swap
system
There's no data in mpp-solar that I recognize other than the machine name and the command I sent: 
1714998949199141300  QID     RebuiltMQTT   00002111080160 mpp-solar
There aren't any command responses, although responses do show when I monitor the broker and send QPIGS, for example.

To use another data source for MQTT, what would I name the topic in order for it to be processed by influx?

Thanks!

**Reply by jblance** [reply]
post the mqtt messages you are seeing, eg `$ mosquitto_sub -h localhost -v -t "#"` results
what is your telegraf config?

> There's no data in mpp-solar that I recognize other than the machine name and the command I sent:
> 1714998949199141300 QID RebuiltMQTT 00002111080160 mpp-solar

This the result of the `mpp-solar -q localhost -c QID -o influx2_mqtt` command, what other commands have you run
What is the data you dont recognize?

remember telegraf / influx / grafana are not part of mppsolar, so have their own support paths (I also dont use them any more), so while I can provide some hints you might be better asking in their own forums - all we are doing from mppsolar is sending mqtt messages.
Telegraf is configured to read mqtt messages and populate influx and later grafana uses influx as a data source to display

Is very hard to help without detailed information - include all commands that you run, any output, any logs / results from each of the components (each of which will have the ability to increase the verbosity of logging), eg see here for telegraf https://docs.influxdata.com/telegraf/v1/configure_plugins/troubleshoot/

**Reply by bobcov** [reply]
I didn't realize that these other elements are not part of your official setup. This is an important perspective I was missing. 
Because of the size of Telegraf.conf, it might be easier to say what I should have there. I followed the instructions to make sure [[outpus.influxdb]] was commented out and I created the mqtt-input.conf file as directed. I also created the file influx-output.conf.

Looking at what is being sent to the broker and then looking at influx, for some commands, such as QFLAGS, I see the word "disabled" which was a collected value in the response, but for other commands such as QPIGS, I don't get any values even though the appear on the broker:
mpp-solar mpp-solar,command=QID serial_number="00002111080160"
mpp-solar mpp-solar,command=QPIGS ac_input_voltage=0.0
mpp-solar mpp-solar,command=QPIGS ac_input_frequency=0.0
mpp-solar mpp-solar,command=QPIGS ac_output_voltage=230.2
mpp-solar mpp-solar,command=QPIGS ac_output_frequency=50.0
mpp-solar mpp-solar,command=QPIGS ac_output_apparent_power=22
mpp-solar mpp-solar,command=QPIGS ac_output_active_power=1


It is not fair of me to ask you to jump into this, so I will stick with mpp-solar questions and with the exception of telegraf.conf, start looking at the other resources you mentioned for answers unless you see something obviously missed.. 
One MPP question is that I think I must have chosen the wrong package option to install because I do not see /etc/mpp-solar directory, but I do see /home/xxxxxx/mpp-solar/docker/dev/config/    Should I reinstall and if so, what is the appropriate command line?
The second question has to do with how to get the program to use python3.12.3 instead of 3.8.10.  I put this line "#!/usr/bin/env python3.12.3" at the top of  ~/.local/lib/python3.8/site-packages/mppsolar/__init__.py   [the file name has been truncated by this message board] Was that the right thing to do? I think it might have worked because yesterday, the -o option was failing, but it works now.  But I did so many things, I cannot be sure if this was the problem I was trying to solve when I added that line to init.py

Thank you much. With the exception of my mistakes, your program seems to be functioning fine.

**Reply by jblance** [reply]
You need to create the /etc/mpp-solar directory yourself
You need to make sure to install the package using the pip that relates to the version of python needed otherwise pyenv is a tool that can help switching between versions
I think you need to check the telegraf logs to see if it's having issues with some of the mqtt messages
